### PR TITLE
test(bin-shared): improve network notifier test

### PR DESCRIPTION
**Pull request by ReactorScram**
On Windows, the network notifier always notifies once at startup. We make the DNS notifier and Linux match this behavior, and we assert it in the unit test.

Part of a yak shave towards removing Tauri.

### CODE REVIEW

File: rust/bin-shared/src/network_changes/linux.rs [hunk 1]
```diff
@@ -69,7 +72,12 @@ pub async fn new_network_notifier(
     }
 }
 
-pub enum Worker {
+pub struct Worker {
+    just_started: bool,
+    inner: Inner,
```
Comment (ReactorScram) [inline rust/bin-shared/src/network_changes/linux.rs:77]: Needed to notify once at startup

File: rust/bin-shared/src/network_changes/windows.rs [hunk 2]
```diff
@@ -552,27 +575,43 @@ mod async_dns {
...
             .ok()
             .context("`RegNotifyChangeKeyValue` failed")?;
             Ok(())
         }
+
+        fn close(mut self) -> Result<()> {
+            self.close_dont_drop()
+        }
+
+        fn close_dont_drop(&mut self) -> Result<()> {
```
Comment (thomaseizinger) [inline rust/bin-shared/src/network_changes/windows.rs:610]: Can you document what is happening here?

File: rust/bin-shared/tests/network_notifiers.rs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        .unwrap()
+        .unwrap();
+
+    // After that first notification, we shouldn't get any other notifications during a normal unit test.
+
+    let waker = noop_waker();
+    let mut ctx = Context::from_waker(&waker);
+
+    assert!(matches!(pin!(dns.notified()).poll(&mut ctx), Poll::Pending));
+    assert!(matches!(pin!(net.notified()).poll(&mut ctx), Poll::Pending));
```
Comment (thomaseizinger) [inline rust/bin-shared/tests/network_notifiers.rs]: You might find this one handy: https://docs.rs/futures/latest/futures/future/trait.FutureExt.html#method.now_or_never

**Review by thomaseizinger — APPROVED** [review]
Nice work!

**Pull request merged**